mrp_bom_product_price_margin/views/view_mrp_bom.xml

```diff
@@ -0,0 +1,151 @@
+                        name="product_uom_id"
+                        readonly="1"
+                        style="pointer-events:none;"
+                    />
+              </div>
+            </xpath>
+
+            <!-- BoM Lines -->
+            <xpath
+                expr="//field[@name='bom_line_ids']/tree/field[@name='product_uom_id']"
+                position="before"
+            >
+                <field name="currency_id" invisible="1" />
+                <field name="standard_price_unit" widget="monetary" />
+                <field
+                    name="standard_price_subtotal"
+                    readonly="1"
+                    sum="Total"
+                    widget="monetary"
+                />
+                <field
+                    name="standard_price_subtotal_percentage"
+                    string="ST %"
+                    widget="percentage"
+                />
+            </xpath>
+
+        </field>
+    </record>
+
+    <!-- ************************************************************ -->
+    <!-- MRP Bill of Materials - Tree -->
+    <!-- ************************************************************ -->
+
+    <record id="view_mrp_bom_tree" model="ir.ui.view">
+        <field name="model">mrp.bom</field>
+        <field name="inherit_id" ref="mrp.mrp_bom_tree_view" />
+        <field name="arch" type="xml">
+            <xpath expr="//field[@name='code']" position="after">
+                <field
+                    name="diff_product_bom_standard_price"
+                    string="Diff BoM and Product"
+                    invisible="1"
+                />
+            </xpath>
+            <xpath expr="//tree" position="attributes">
+                <attribute
+                    name="decoration-warning"
+                >diff_product_bom_standard_price != False</attribute>
+            </xpath>
+            <xpath expr="//tree" position="attributes">
+                <attribute
+                    name="decoration-danger"
+                >product_margin_rate_percentage &lt; 0</attribute>
+            </xpath>
+            <xpath expr="//field[@name='product_tmpl_id']" position="after">
+                <field name="currency_id" invisible="1" />
+                <field name="standard_price" widget="monetary" />
+                <button
+                    name="set_product_standard_price"
+                    icon="fa-arrow-right"
+                    string="Update Product standard price"
+                    type="object"
+                    attrs="{'invisible': [('diff_product_bom_standard_price', '=', False)]}"
+                    help="It will set Product's standard price with this price"
+                />
+                <field name="product_standard_price" widget="monetary" />
+                <field name="product_sale_price" widget="monetary" />
+                <field name="product_margin_rate_percentage" widget="percentage" />
+            </xpath>
```

Comment:
All that new fields are interested to manage massively all the margin. However it adds a lot of column, that can  be useless in everyday life, and that overloads the screen.

I suggest to set all the field in optional="hide". What do you think ?